Review of the scaling documentation (#6706)
Particular/docs.particular.net: nservicebus/scaling.md

redirects:
 - transports/scale-out
 - nservicebus/architecture/scaling
reviewed: 2021-07-23
reviewed: 2024-06-17
---

This page describes how to scale out endpoints using NServiceBus. There are several reasons to scale out, such as to achieve higher message throughput, or to provide high availability.

> It is important to distinguish between a logical endpoint and an endpoint instance. Review the documentation on [logical endpoints](/nservicebus/endpoints/) for more details.

Reviewer: reason for removing?

Author: I was going back and forth on this one but ultimately felt that this is schooling people and I don't like to be schooled. So I figured by linking to the appropriate definition, it should be good enough for people to follow through. I'm also not opposed to bringing back the wording if we prefer that.

> There are several reasons to scale out [logical endpoints](/nservicebus/endpoints/), such as to achieve higher message throughput, or to provide high availability to name a few.

Reviewer: technically "a few" is 3 or more, so this is naming a couple

> There are several reasons to scale out [logical endpoints](/nservicebus/endpoints/), such as to achieve higher message throughput, or to provide high availability to name a few.

Author: ```suggestion
There are several reasons to scale out [logical endpoints](/nservicebus/endpoints/), such as to achieve higher message throughput, or to provide high availability to name a couple.
```

> There are several reasons to scale out [logical endpoints](/nservicebus/endpoints/), such as to achieve higher message throughput, or to provide high availability to name a couple.

Author: ```suggestion
There are several reasons to scale out [logical endpoints](/nservicebus/endpoints/), such as to achieve higher message throughput, or to provide high availability.
```

If message throughput is an issue, the first method to consider is splitting up [message handlers](/nservicebus/handlers/) and [sagas](/nservicebus/sagas/) over multiple logical endpoints.

One message type might take considerably longer to process than other message types. The faster processing messages might suffer in throughput because of the slower processing messages. A good way to monitor and detect this is by using [ServicePulse's monitoring capabilities](/monitoring/metrics/in-servicepulse.md).

> Separating slower messages from faster messages leads to higher throughput for the faster messages. For this reason, it can be beneficial to include messages and/or handlers in separate assemblies, making it easier to separate them from others.

Author: > For this reason, it can be beneficial to include messages and/or handlers in separate assemblies, making it easier to separate them from others.

I have a hard time following this reasoning.

> Separating slower messages from faster messages leads to higher throughput for the faster messages. For this reason, it can be beneficial to include messages and/or handlers in separate assemblies, making it easier to separate them from others.

Author: ```suggestion
Separating slower messages from faster messages leads to higher throughput for the faster messages. For this reason, it can be beneficial to seperate different message types and their corresponding handlers into different logical endpoints.
```
